syntax_sugar/infix.py

```diff
@@ -39,18 +39,26 @@ def __init__(self, start, end):
 
         self.start = start
         self.curr = self.start
-        self.step = 1 if end > start else -1
+        self._step = 1 if end > start else -1
         self.end = end
 
+    @property
+    def step(self):
+        return self._step
+
+    @step.setter
+    def step(self, value):
+        if value == 0 or not value /is_a/ int:
+            raise TypeError('Interval must be an integer different from 0')
```

Comment:
I think we still need the code to `raise ValueError('Cannot define increasing ranges with negative step')` in order to detect potential errors early.